### ISSUE TITLE
The time magic bug comes back with ipython 7.6.0

### ISSUE THREAD

**Issue by daxiongshu** [ipython/ipython#11813]
I run into the exact same error as stated in this closed issue https://github.com/ipython/ipython/issues/11659 but with ipython 7.6.0. Downgrading to ipython 7.5.0 fixes it for me.

**Comment by wenjuno** [comment]
I didn't run into the exact same issue. It appears to me that the problem is that if there are multiple statements in the %%time cell and the last statement is an assignment, the last statement is skipped.
Here are some examples (python 3.6):

```python
In [1]: def sleep_a_sec(n):
   ...:     import time
   ...:     time.sleep(1)
   ...:     return n
   ...:

In [2]: %%time
   ...: a = sleep_a_sec(10)
   ...:
   ...:
CPU times: user 1.25 ms, sys: 1.26 ms, total: 2.5 ms
Wall time: 1 s

In [3]: a
Out[3]: 10

In [4]: %%time
   ...: b = 1
   ...: c = sleep_a_sec(20)
   ...:
   ...:
CPU times: user 4 µs, sys: 0 ns, total: 4 µs
Wall time: 8.82 µs

In [5]: b
Out[5]: 1

In [6]: c
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-6-2b66fd261ee5> in <module>
----> 1 c

NameError: name 'c' is not defined

In [7]: %%time
   ...: b = 11
   ...: sleep_a_sec(21)
   ...:
   ...:
CPU times: user 962 µs, sys: 1.12 ms, total: 2.08 ms
Wall time: 1 s
Out[7]: 21

In [8]: b
Out[8]: 11

In [9]: %%time
   ...: b = 101
   ...: c = 102
   ...: d = sleep_a_sec(21)
   ...:
   ...:
CPU times: user 4 µs, sys: 1e+03 ns, total: 5 µs
Wall time: 9.06 µs

In [10]: b
Out[10]: 101

In [11]: c
Out[11]: 102

In [12]: d
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-12-e983f374794d> in <module>
----> 1 d

NameError: name 'd' is not defined
```

**Comment by Carreau** [comment]
Please upgrade to 7.6.1 and retry.

**Comment by aakhmetz** [comment]
I have another issue with ```%%time``` as
```python
AttributeError                            Traceback (most recent call last)
<ipython-input-42-6bcdd5a80f02> in <module>
----> 1 get_ipython().run_cell_magic('time', '', 'import csv\n')

~/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py in run_cell_magic(self, magic_name, line, cell)

</home/aakhmetz/anaconda3/lib/python3.6/site-packages/decorator.py:decorator-gen-61> in time(self, line, cell, local_ns)

~/anaconda3/lib/python3.6/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)

~/anaconda3/lib/python3.6/site-packages/IPython/core/magics/execution.py in time(self, line, cell, local_ns)

AttributeError: 'If' object has no attribute 'value'
```

@Carreau, would it be also fixed with 7.6.1? Thanks

**Comment by Carreau** [comment]
> would it be also fixed with 7.6.1? Thanks

Well you tell me; you know the code you enter that failed; I don't.

**Comment by wenjuno** [comment]
@Carreau It works for my use case. Thanks for the prompt fix!
